azext_iot/tests/test_iot_pnp_int.py

```diff
@@ -0,0 +1,183 @@
+from azure.cli.testsdk import LiveScenarioTest
+from azure.cli.core.util import read_file_content
+
+
+# Add test tools to path
+sys.path.append(os.path.abspath(os.path.join('.', 'iotext_test_tools')))
+
+# Set these to the proper PnP Endpoint, PnP Cstring and PnP Repository for Live Integration Tests.
+_endpoint = os.environ.get('azext_pnp_endpoint')
+_repo_id = os.environ.get('azext_pnp_repository')
+_repo_cs = os.environ.get('azext_pnp_cs')
+
+_interface_payload = 'test_pnp_create_payload_interface.json'
+_capability_model_payload = 'test_pnp_create_payload_model.json'
+
+if not all([_endpoint, _repo_id, _repo_cs]):
+    raise ValueError('Set azext_pnp_endpoint, azext_pnp_repository and azext_pnp_cs to run PnP model integration tests.')
+
+
+def change_dir():
+    from inspect import getsourcefile
+    os.chdir(os.path.dirname(os.path.abspath(getsourcefile(lambda: 0))))
+
+
+class TestPnPModel(LiveScenarioTest):
+
+    LiveScenarioTest.rand_val = random.randint(1, 10001)
+
+    def __init__(self, _):
+        from iotext_test_tools import DummyCliOutputProducer
+        super(TestPnPModel, self).__init__(_)
+        self.cli_ctx = DummyCliOutputProducer()
+        self.kwargs.update({
+            'endpoint': _endpoint,
+            'repo': _repo_id,
+            'repo_cs': _repo_cs,
+            'interface': 'test_interface_definition.json',
+            'interface-updated': 'test_interface_updated_definition.json',
+            'model': 'test_model_definition.json',
+            'model-updated': 'test_model_updated_definition.json'
+        })
+
+    def setUp(self):
+        change_dir()
+        if(self._testMethodName == 'test_interface_life_cycle'):
+            interface = str(read_file_content(_interface_payload))
+            _interface_id = '{}{}'.format(json.loads(interface)['@id'], LiveScenarioTest.rand_val)
+            self.kwargs.update({'interface_id': _interface_id})
+            interface_newContent = interface.replace(json.loads(interface)['@id'], self.kwargs['interface_id'])
+            interface_newContent = interface_newContent.replace('\n', '')
+
+            fo = open(self.kwargs['interface'], "w+", encoding='utf-8')
+            fo.write(interface_newContent)
+            fo.close()
+
+        if(self._testMethodName == 'test_model_life_cycle'):
+            model = str(read_file_content(_capability_model_payload))
+            _model_id = '{}{}'.format(json.loads(model)['@id'], LiveScenarioTest.rand_val)
+            self.kwargs.update({'model_id': _model_id})
+            model_newContent = model.replace(json.loads(model)['@id'], self.kwargs['model_id'])
+            model_newContent = model_newContent.replace('\n', '')
+
+            fo = open(self.kwargs['model'], "w+", encoding='utf-8')
+            fo.write(model_newContent)
+            fo.close()
+
+    # TODO: @anusapan - Maybe put a helper like this in the shared lib, when you create it?
+    def command_execute_assert(self, command, asserts):
```

Comment:
It looks like you are not using this particular helper. Safe to delete?

Reply:
Removed